modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/viewmodel/OnboardingUpgradeFeaturesViewModel.kt

```diff
@@ -54,36 +50,25 @@ class OnboardingUpgradeFeaturesViewModel @Inject constructor(
     private val showPatronOnly = savedStateHandle.get<Boolean>("show_patron_only")
 
     init {
-        if (FeatureFlag.isEnabled(Feature.ADD_PATRON_ENABLED)) {
-            viewModelScope.launch {
-                subscriptionManager
-                    .observeProductDetails()
-                    .asFlow()
-                    .stateIn(viewModelScope)
-                    .collect { productDetails ->
-                        val subscriptions = when (productDetails) {
-                            is ProductDetailsState.Error -> null
-                            is ProductDetailsState.Loaded -> productDetails.productDetails.mapNotNull { productDetailsState ->
-                                Subscription.fromProductDetails(
-                                    productDetails = productDetailsState,
-                                    isFreeTrialEligible = subscriptionManager.isFreeTrialEligible(
-                                        SubscriptionMapper.mapProductIdToTier(productDetailsState.productId),
-                                    ),
-                                )
-                            }
-                        } ?: emptyList()
-                        updateState(subscriptions)
-                    }
-            }
-        } else {
-            val accessibiltyManager = getApplication<Application>().getSystemService(Context.ACCESSIBILITY_SERVICE)
-                as? AccessibilityManager
-
-            var isTouchExplorationEnabled = accessibiltyManager?.isTouchExplorationEnabled ?: false
-            accessibiltyManager?.addTouchExplorationStateChangeListener {
-                isTouchExplorationEnabled = it
-            }
-            _state.update { OnboardingUpgradeFeaturesState.OldLoaded(isTouchExplorationEnabled) }
+        viewModelScope.launch {
+            subscriptionManager
+                .observeProductDetails()
+                .asFlow()
+                .stateIn(viewModelScope)
+                .collect { productDetails ->
+                    val subscriptions = when (productDetails) {
+                        is ProductDetailsState.Error -> null
+                        is ProductDetailsState.Loaded -> productDetails.productDetails.mapNotNull { productDetailsState ->
+                            Subscription.fromProductDetails(
+                                productDetails = productDetailsState,
+                                isFreeTrialEligible = subscriptionManager.isFreeTrialEligible(
+                                    SubscriptionMapper.mapProductIdToTier(productDetailsState.productId),
+                                ),
+                            )
+                        }
+                    } ?: emptyList()
```

Comment:
I go back and forth between early returns vs using null and handling it later. If either of you have any thoughts about why one approach is better than another I'd be interested in learning what you think. 🙂

Reply:
I don't think I have any general principle that guides me here. It all depends on the context. I suggested it here because it requires less thinking from me. When I see `?: foo` I usually need to scan code where it can be `null`. If there were more cases, like this

```kotlin
val value = when (check) {
  FOO -> mightReturnNull1()
  BAR -> mightReturnNull2()
  BAZ -> listOf(1, 2, 3)
  KUX -> mightReturnNull3()
}
```

I'd probably go for the Elvis operator. But if they were inlined operations I'd probably do what I suggested here.

Reply:
For me I like the idea to return `emptyList()` for `ProductDetailsState.Error` for the same reason @MiSikora pointed. It's easier to understand